packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -0,0 +1,547 @@
+      return { outOfBandRecord, invitation: convertToOldInvitation(outOfBandRecord.outOfBandMessage) }
+    }
+    throw new AriesFrameworkError(
+      `Invalid value of handshakeProtocols in config. Value is ${config.handshakeProtocols}, but this method supports only ${HandshakeProtocol.Connections}.`
+    )
+  }
+
+  /**
+   * Parses URL, decodes invitation and calls `receiveMessage` with parsed invitation message.
+   *
+   * Agent role: receiver (invitee)
+   *
+   * @param invitationUrl url containing a base64 encoded invitation to receive
+   * @param config configuration of how out-of-band invitation should be processed
+   * @returns out-of-band record and connection record if one has been created
+   */
+  public async receiveInvitationFromUrl(invitationUrl: string, config: ReceiveOutOfBandMessageConfig = {}) {
+    const message = await this.parseInvitation(invitationUrl)
+    return this.receiveInvitation(message, config)
+  }
+
+  /**
+   * Parses URL containing encoded invitation and returns invitation message.
+   *
+   * @param invitationUrl URL containing encoded invitation
+   *
+   * @returns OutOfBandMessage
+   */
+  public async parseInvitation(invitationUrl: string) {
+    const parsedUrl = parseUrl(invitationUrl).query
+    if (parsedUrl['oob']) {
+      const outOfBandMessage = await OutOfBandMessage.fromUrl(invitationUrl)
+      return outOfBandMessage
+    } else if (parsedUrl['c_i'] || parsedUrl['d_m']) {
+      const invitation = await ConnectionInvitationMessage.fromUrl(invitationUrl)
+      return convertToNewInvitation(invitation)
+    }
+    throw new AriesFrameworkError(
+      'InvitationUrl is invalid. It needs to contain one, and only one, of the following parameters: `oob`, `c_i` or `d_m`.'
+    )
+  }
+
+  /**
+   * Creates inbound out-of-band record and assings out-of-band invitation message to it if the
+   * message is valid. It automatically passes out-of-band invitation for further processing to
+   * `acceptInvitation` method. If you don't want to do that you can set `autoAcceptInvitation`
+   * attribute in `config` parameter to `false` and accept the message later by calling
+   * `acceptInvitation`.
+   *
+   * It supports both OOB (Aries RFC 0434: Out-of-Band Protocol 1.1) and Connection Invitation
+   * (0160: Connection Protocol).
+   *
+   * Agent role: receiver (invitee)
+   *
+   * @param outOfBandMessage
+   * @param config config for handling of invitation
+   *
+   * @returns out-of-band record and connection record if one has been created.
+   */
+  public async receiveInvitation(
+    outOfBandMessage: OutOfBandMessage,
+    config: ReceiveOutOfBandMessageConfig = {}
+  ): Promise<{ outOfBandRecord: OutOfBandRecord; connectionRecord?: ConnectionRecord }> {
+    const { handshakeProtocols } = outOfBandMessage
+    const { routing } = config
+
+    const autoAcceptInvitation = config.autoAcceptInvitation ?? true
+    const autoAcceptConnection = config.autoAcceptConnection ?? true
```

Comment:
Should this be `true`? It's the opposite of what we've done in the past where the default is always false.

In addition, we should maybe also take the agentConfig autoAcceptInivtation into account here?

Reply:
I think this one is important to change, or we should otherwise at least be very clear in documentation that this is the inverse of what we've done in the past. Any comments on this one @jakubkoci?